src/core/cast.ts

```diff
@@ -0,0 +1,131 @@
+import type { CastFunction, CastFunctions, CastType, CastedValue } from '../types.js'
+import { isFunction, isString, setValue } from '../utils/index.js'
+
+const castFunctions: CastFunctions = {}
```

Comment:
It's probably better to keep it simple and let typescript do its inference magic here. Just declare and assign `castFunctions` in one shot.

Current
![image](https://github.com/user-attachments/assets/0186e0c3-1532-4eb7-9e1b-96fb13f04d75)

With inference
![image](https://github.com/user-attachments/assets/df6aa8f5-a2d4-479d-88fb-09f574829166)

Reply:
Anf if you still need CastType, same logic, use `keyof typeof`: The code is right and the typing is inferred
![image](https://github.com/user-attachments/assets/44e90ad5-c6b0-4ba2-8785-b8cef20da48c)

Reply:
Same thing if you ever need a specific return type of an instance of generic function by the way. You don't need it here I thing but it's good to know 😄 
![image](https://github.com/user-attachments/assets/a88ba2a4-f65a-42ae-82fb-1337cceaa1e3)

Reply:
There might be other cases like that, I'll let you check. Just remember that the simplest approach is often the best with TS since the inference is often more powerful than a complex typing.